oak/client/application_client.h

```diff
@@ -94,11 +96,18 @@ class ApplicationClient {
 
     NonceGenerator<kPerChannelNonceSizeBytes> nonce_generator;
     auto channel_authorization_token_bytes = NonceToBytes(nonce_generator.NextNonce());
+
+    std::string channel_authorization_token_hmac =
+        oak::utils::hmac_sha256(channel_authorization_token_bytes, kBearerTokenHmacData)
+            .ValueOrDie();
+
+    // Create composite channel credentials by using the (secret) token to authenticate, and the
+    // (public) token HMAC to set a corresponding policy on the data.
     auto call_credentials = grpc::CompositeCallCredentials(
         grpc::MetadataCredentialsFromPlugin(
             absl::make_unique<AuthorizationBearerTokenMetadata>(channel_authorization_token_bytes)),
         grpc::MetadataCredentialsFromPlugin(absl::make_unique<PolicyMetadata>(
-            AuthorizationBearerTokenPolicy(channel_authorization_token_bytes))));
+            AuthorizationBearerTokenPolicy(channel_authorization_token_hmac))));
```

Comment:
Does this really have to be passed as part of the call credentials? If the label should apply to the entire call, the server can just recreate it from the auth token bytes that are already provided.

Also, even though it is out of scope of ot this PR, it seems feasible that a single call might contain multiple pieces of data that each require their own label.

Reply:
In the case where the client passes a label that is different from the users own label, it feels inapprotiate to use CallCredentials as a mechanism to pass such a label. It is a convenient mechanism for getting something into the grpc metadata, but overloads the use of credentials in a way that is likely to cause confusion.

Reply:
That's a good point, do you think we should make it more explicit by making it an explicit argument to each invocation instead? Currently the code here is assuming the label is set once at the client level and never changes, but obviously that does not scale.

Reply:
I think it would be a good idea, but can probably be done as part of the work to support an IDL that will handle labels more naturally.